File: src/main/kotlin/com/yzzzzun/playground/basic/java/Penguin.java
package com.yzzzzun.playground.basic.java;

public class Penguin extends Animal implements Swimable, Flyable {

	private final int wingCount;

	public Penguin(String species) {
		super(species, 2);
		this.wingCount = 2;
	}

	@Override
	public void move() {
		System.out.println("팽귄 move");
	}

	@Override
	public int getLegCount() {
		return super.legCount + this.wingCount;
	}

	@Override
	public void act() {
		Swimable.super.act();
		Flyable.super.act();
	}
}

File: src/main/kotlin/com/yzzzzun/playground/basic/kotlin/Penguin.kt
package com.yzzzzun.playground.basic.kotlin

class Penguin(species: String) : Animal(species, 2), Swimable, Flyable {

    private val wingCount: Int = 2

    override fun move() {
        println("팽귄 move")
    }

    override val legCount: Int
        get() = super.legCount + this.wingCount

    override fun act() {
        super<Swimable>.act()
        super<Flyable>.act()
    }
}
